rllib/core/optim/tests/test_rl_optimizer.py

```diff
@@ -0,0 +1,378 @@
+import gym
+import pytest
+import numpy as np
+import torch
+from typing import Any, List, Mapping
+
+from ray.rllib.algorithms import AlgorithmConfig
+from ray.rllib.offline import IOContext
+from ray.rllib.offline.dataset_reader import (
+    DatasetReader,
+    get_dataset_and_shards,
+)
+
+import ray
+from ray.rllib.core.examples.simple_ppo_rl_module import FCNet, FCConfig
+from ray.rllib.core.optim.rl_optimizer import RLOptimizer
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.torch.torch_rl_module import TorchRLModule
+from ray.rllib.core.rl_module.torch.tests.test_torch_rl_module import (
+    to_numpy,
+    to_tensor,
+)
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.models.specs.specs_dict import ModelSpec
+from ray.rllib.models.specs.specs_torch import TorchTensorSpec
+from ray.rllib.models.torch.torch_distributions import (
+    TorchDeterministic,
+    TorchCategorical,
+)
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.test_utils import check
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+
+
+def model_norm(model: torch.nn.Module) -> float:
+    """Compute the norm of the weights of a model.
+
+    Args:
+        model: The model whose weights to compute a norm over.
+
+    Returns:
+        the norm of model.
+    """
+    total_norm = 0
+    for p in model.parameters():
+        param_norm = p.detach().data.norm(2)
+        total_norm += param_norm.item() ** 2
+    total_norm = total_norm**0.5
+    return total_norm
+
+
+def do_rollouts(
+    env: gym.Env, module_for_inference: RLModule, num_rollouts: int
+) -> SampleBatch:
+    _returns = []
+    _rollouts = []
+    for _ in range(num_rollouts):
+        obs = env.reset()
+        _obs, _next_obs, _actions, _rewards, _dones = [], [], [], [], []
+        _return = -0
+        for _ in range(env._max_episode_steps):
+            _obs.append(obs)
+            fwd_out = module_for_inference.forward_inference(
+                {"obs": to_tensor(obs)[None]}
+            )
+            action = to_numpy(fwd_out["action_dist"].sample().squeeze(0))
+            next_obs, reward, done, _ = env.step(action)
+            _next_obs.append(next_obs)
+            _actions.append([action])
+            _rewards.append([reward])
+            _dones.append([done])
+            _return += reward
+            if done:
+                break
+            obs = next_obs
+        batch = SampleBatch(
+            {
+                "obs": _obs,
+                "next_obs": _next_obs,
+                "actions": _actions,
+                "rewards": _rewards,
+                "dones": _dones,
+            }
+        )
+        _returns.append(_return)
+        _rollouts.append(batch)
+    return np.mean(_returns), _returns, _rollouts
+
+
+class BCTorchModule(TorchRLModule):
```

Comment:
@sven1977 @gjoliver @avnishn  How do you feel about the transparency of this module? 
You know exactly what the input / output fmt of each api call is without needing to read its code or documentation, if you know the design pattern of RLModules.

Reply:
the only one I thought had a friction point was when defining input spec functions.

My thought process was that it was easy to miss that I needed to implement these functions because some of them had default keys to be filtered out and if I didn't know about the filtering I wouldn't understand that I need to extend these functions in order for additional keys need to appear in the filtered batch.

Reply:
I see. I have turned off the filtering by default now in the main integration PR.